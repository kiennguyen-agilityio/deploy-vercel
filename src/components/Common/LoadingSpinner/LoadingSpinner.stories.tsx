import { StoryObj, Meta } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';

import LoadingSpinner from './LoadingSpinner';

export default {
  title: 'Components/LoadingSpinner',
  component: LoadingSpinner,
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    ),
  ],
} as Meta;

type Story = StoryObj<typeof LoadingSpinner>;

export const Default: Story = {};
