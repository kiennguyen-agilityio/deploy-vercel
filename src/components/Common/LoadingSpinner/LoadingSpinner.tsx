import { memo } from 'react';
import './LoadingSpinner.css';

const LoadingSpinner = () => (
  <div className="spinner-container">
    <div className="loading-spinner" />
  </div>
);

export default memo(LoadingSpinner);
