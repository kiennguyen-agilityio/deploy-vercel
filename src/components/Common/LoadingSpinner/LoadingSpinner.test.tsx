import { render } from '@testing-library/react';
import LoadingSpinner from './LoadingSpinner';

describe('Testing spinner', () => {
  it('Should render component', () => {
    const { container } = render(<LoadingSpinner />);

    expect(container).toMatchSnapshot();
  });
});
