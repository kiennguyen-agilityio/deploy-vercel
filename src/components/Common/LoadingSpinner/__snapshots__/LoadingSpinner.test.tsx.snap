// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing spinner Should render component 1`] = `
<div>
  <div
    class="spinner-container"
  >
    <div
      class="loading-spinner"
    />
  </div>
</div>
`;
